lib/tarpon/client.rb

```diff
@@ -1,19 +1,33 @@
 # frozen_string_literal: true
 
+require 'forwardable'
 require 'tarpon/configuration'
 
 module Tarpon
   class Client
-    extend Configuration
+    include Configuration
+
+    def self.default
+      @default ||= new
+    end
+
+    def initialize(**config, &block)
```

Comment:
It feels like the library usage becomes a bit more confusing when providing multiple ways to configure it, especially to support a use case that is probably not common.

Do you think your approach could fit in the `.configure` method? And it's perfectly fine for `.configure` to return the instance it creates.

I played with your branch a little bit and here's an idea of what the client looks like:
```ruby
  class Client
    include Configuration

    class << self
      extend Forwardable
      attr_accessor :default

      def configure(&block)
        default_instance = new
        default_instance.class.default = default_instance
        block&.call(default_instance)
        default_instance
      end

      def_delegators :default, *Configuration.public_instance_methods(true)
      def_delegators :default, :subscriber, :receipt
    end

    def subscriber(app_user_id)
      Request::Subscriber.new(app_user_id: app_user_id)
    end

    def receipt
      Request::Receipt.new
    end
  end
```

Brought the `.configure` method to the `Client` in order to make the delegators work now that we include the `Configuration` instead of extending it, so the `Configuration` module is left only with the possible attributes to be set.

What do you think?

Reply:
I'm not super big on `.configure` overriding the default like that—I think it adds too many responsibilities to the method.

One other thing I wouldn't be super happy about is that this API makes `.configure` a required call in order for the library to work. If at any point you wanted to add "blessed" environment variables so that, for example, `REVENUE_CAT_PUBLIC_API_KEY` and `REVENUE_CAT_PRIVATE_API_KEY` are automatically captured without any need for setup from the user—which is a fairly common practice in lots of libraries, this wouldn't quite work (since `.default` would, by default, return `nil`).

That said, I don't feel too strongly about this, and I'm happy to accommodate whatever you think it's best.

The only thing that I would ask is if it makes more sense for you that `.configure` overwrites the default every time it's called, vs using `||=` and only assigning a default instance once.

(In practice this is probably moot, as people that need more than one configuration will likely not be using `Client.default`, but trying to keep the design clean 😅)

Reply:
>I'm not super big on .configure overriding the default like that—I think it adds too many responsibilities to the method.

I hear you, such a fundamental change on the library internals makes it difficult to support both ways of using it without breaking some principles.

Will be accepting the PR as it is, I think in the near future the `.configure` should be deprecated or at least work as an alias for instantiating the client, becoming the only way of using it.